Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,223 @@
+    case phone
+    case video
+    case faceToFace = "face to face"
+}
+```
+At the beginning, the `ConsultationMethod` had indeed 2 cases. This may be trivial here without big consequences however that not always a case and a lie may cost us many of hours.
+
+"Cross-files" comments are even more dangerous to become a lie at some point. Consider the following comment:
+```
+/// Make the booking confirmation screen. The created screen would always refetch
+/// the appointment from the backend.
+public func makeBookingConfirmation(
+    with appointmentId: AppointmentDTO.ID,
+    presentingOn flow: Flow,
+    dismiss: @escaping () -> Void
+) -> UIViewController 
+```
+Such a comment is a leak of information from another file into Builder. Builder doesn't know how BusinessController works. A change in a BusinessController can change the behavior of given screen, while there won't be a need to even enter the builder to find out there is a comment. In the end, comment stays there, unchanged.
+
+In our Pull Requests I've seen sometimes unnecessary comments
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        /// waiting for payment
+        case pending
+        /// user failed to pay for appoitment withing 10 min after the booking
+        case timedOut = "timed_out"
+        /// canceled by user or doctor
+        case cancelled
+        /// paid by user
+        case paid
+        /// doctor finished adding notes on the portal
+        case completed
+        /// patient did not answer the doctor's call
+        case noShow = "no_show"
+    }
+}
+```
+Can be simplified without any lose in understanding what the code represents to:
+
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        /// waiting for payment
+        case waitingForPayment = "pending"
+        /// timeout == 10 min
+        case paymentTimedOut = "timed_out"
+        case cancelledByUserOrDoctor = "cancelled"
+        case paidByUser = "paid"
+        case doctorFinishedAddingNotesOnThePortal = "completed"
+        case patientDidntAnswerTheCall = "no_show"
+    }
+}
+```
+
+## Proposed solution
+
+### Our approach 
+The most important thing is to improve the definition of our approach to comments as currently it isn't said clearly. Moreover, it would be nice to have a common way how we comment our code when we are forced to add them. 
+
+In my opinion, our approach should insist on writing always a **self-documented code**. It should be our goal. However, writing self-documented code **doesn't** mean we shouldn't use comments. It means, that commenting our code is a **necessary evil**:
+
+> The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures. We must have them because we cannot always figure out how to express ourselves without them, but their use is not a cause for celebration.
```

Comment:
Nice 👍

Reply:
I disagree with this paragraph 100% 👎  Comments are *not* a failure. Comments are *not* a necessary evil. Good comments are a *critical* component of writing good code. That said, not all code needs to be commented, and the existing style guide explains when they are needed: explain the WHY, not the HOW.

Reply:
@mluisbrown do you agree with a statement that code is better than a comment and if you can express yourself with code clearly enough then comment is not needed?

Reply:
@TheAdamBorek yes, of course, if you can make your code clear enough then a comment is not needed. It should be rare that we need comments to explain *WHAT* the code does, but IMO we don't have enough comments to explain *WHY* the code is doing what it does.

However I think the current comments section of the style guide already says basically this ☝️ so I don't see the need for any expansion on that.